### PR TITLE
Onboard LAADS subset-band-name Service

### PR DESCRIPTION
**Pull request by rooseveltrp**
## Jira Issue ID

LDDS-199

## Description

Onboarding LAADS Variable Subsetter by Band and Name Service.

### Setup Environment

```
$ clone this repo: https://git.earthdata.nasa.gov/projects/LDDS/repos/laads-subset_band_name-service/browse
$ conda create --name subset_band_name python=3.9 --channel conda-forge --channel defaults
$ conda activate subset_band_name
$ pip install -r ./harmony_python_interface/pip_requirements.txt
$ pip install -r ./docs/pip_requirements.txt
$ cd docs
$ jupyter notebook
```

- Open `Test_Harmony_Integration.ipynb`
- Run all the cells.
- The Output Results should say

```
{'Latitude': (('10*nscans:MODIS_SWATH_Type_L1B',
               'Max_EV_frames:MODIS_SWATH_Type_L1B'),
              (-301531136, 1241841664),
              5,
              0),
 'Longitude': (('10*nscans:MODIS_SWATH_Type_L1B',
                'Max_EV_frames:MODIS_SWATH_Type_L1B'),
               (-301531136, 1241841664),
               5,
               1),
 'EV_500_RefSB': (('Band_500M:MODIS_SWATH_Type_L1B',
                   '20*nscans:MODIS_SWATH_Type_L1B',
                   '2*Max_EV_frames:MODIS_SWATH_Type_L1B'),
                  (83886080, -602996736, -1811283968),
                  23,
                  2)}
```

## PR Acceptance Checklist
* [ ] Acceptance criteria met
* [ ] Tests added/updated (if needed) and passing
* [ ] Documentation updated (if needed)

### CODE REVIEW

File: services/harmony/test/service-image-tags.ts
Comment (indiejames) [inline services/harmony/test/service-image-tags.ts:39]: nitpick: all the other services use kabob case instead of snake case

**Comment by indiejames** [comment]
The test instructions aren't working for me. When I run `conda activate subset_band_name` I get an error message: 
```
CondaError: Run 'conda init' before 'conda activate'
```
Running `conda init` does not fix things.
I have your service running locally - is there a curl command I can use to test it instead of the notebook?

File: services/harmony/env-defaults
Comment (indiejames) [inline services/harmony/env-defaults:475]: by convention the env variable names should match the image name, using uppercase and underscores. So `SUBSETNAMEBAND` would be `SUBSET_BAND_NAME`

**Review by indiejames — APPROVED** [review]
I was able to test this locally with the following curl:
```
 curl -Ln -bj "http://localhost:3000/C1260442414-LAADSCDUAT/ogc-api-coverages/1.0.0/collections/EV_500_RefSB/coverage/rangeset/?granuleId=G1260734287-LAADSCDUAT"
```
The job ran successfully and called the service and produced an hdf file, but I'm not sure what the contents of the hdf file should be.

**Review by chris-durbin — COMMENTED** [review]
I'll make changes to our deployment repo and test it all out in a sandbox environment (most likely on Monday).

File: services/harmony/env-defaults
Comment (rooseveltrp) [reply services/harmony/env-defaults:475]: @indiejames it should see the variables/data for longitude, latitude and  EV_500_RefSB.

File: config/services.yml
Comment (rooseveltrp) [reply config/services.yml]: Thank you @chris-durbin and @indiejames! I replaced `SUBSETNAMEBAND` with `SUBSET_NAME_BAND`. Kindly do a pull for the changes.

File: services/harmony/test/service-image-tags.ts
Comment (chris-durbin) [inline services/harmony/test/service-image-tags.ts:37]: TSError: ⨯ Unable to compile TypeScript:
Error: test/service-image-tags.ts(37,3): error TS1117: An object literal cannot have multiple properties with the same name.
Error: test/service-image-tags.ts(38,3): error TS1117: An object literal cannot have multiple properties with the same name.
Error: test/service-image-tags.ts(40,3): error TS1117: An object literal cannot have multiple properties with the same name.

geoloco, batchee, and giovanni-adapter are already defined - looks like they were copied twice

Comment (rooseveltrp) [reply services/harmony/test/service-image-tags.ts]: ah, must've been a bad merge. Removed the duplicates.

@chris-durbin also renamed to `subset_band_name` where applicable.

File: services/harmony/test/service-image-tags.ts
Comment (rooseveltrp) [reply services/harmony/test/service-image-tags.ts]: ah, good catch. Changed to kabob case.

**Review by chris-durbin — APPROVED** [review]
Successfully tested out with a sandbox deployment. I will merge this in once the deployment repo PR is approved.

**Pull request merged**
